src/components/BlockchainList/PostsList.tsx

```diff
@@ -1,36 +1,62 @@
-import { Suspense } from 'preact/compat'
+import { LoadingText } from 'components/Text'
+import { Suspense, useState } from 'preact/compat'
 import { useSnapshot } from 'valtio'
 import BlockchainPost from 'components/BlockchainList/BlockchainPost'
+import InfiniteScroll from 'react-infinite-scroll-component'
 import ListLoading from 'components/ListLoading'
 import NoPosts from 'components/BlockchainList/NoPosts'
 import PostStore from 'stores/PostStore'
+import classnames, { display, flexDirection, gap } from 'classnames/tailwind'
 import useScrollToAnchor from 'hooks/useScrollToAnchor'
 
+const scrollContainer = classnames(
+  display('flex'),
+  flexDirection('flex-col'),
+  gap('gap-y-4')
+)
+
 function BlockchainPostsListSuspended() {
-  const { selectedPosts, selectedToken } = useSnapshot(PostStore)
+  const { selectedPosts } = useSnapshot(PostStore)
   useScrollToAnchor()
 
-  return (
-    <>
-      {!selectedPosts.length ? (
-        <NoPosts />
-      ) : (
-        selectedPosts
-          .filter((post) =>
-            selectedToken ? post.derivativeAddress === selectedToken : post
-          )
-          .map((post) => (
-            <BlockchainPost
-              key={post.id}
-              id={Number(post.id)}
-              timestamp={Number(post.timestamp)}
-              text={post.post}
-              sender={post.sender}
-              derivativeAddress={post.derivativeAddress}
-            />
-          ))
-      )}
-    </>
+  const data = selectedPosts.filter((post) =>
+    PostStore.selectedToken
+      ? post.derivativeAddress === PostStore.selectedToken
+      : post
+  )
+  // Take first 10 posts from the list
+  const [items, setItems] = useState(data.slice(0, 10))
+  const loadedItemsAmount = items.length
+  // If there are more items to be loaded then take data from the list
+  const loadMoreItems = () => {
+    setItems([
+      ...items,
+      ...data.slice(loadedItemsAmount, loadedItemsAmount + 10),
+    ])
+  }
+
+  return data.length ? (
+    <InfiniteScroll
+      endMessage={<></>}
+      next={loadMoreItems}
+      className={scrollContainer}
+      dataLength={loadedItemsAmount}
+      hasMore={loadedItemsAmount < data.length}
+      loader={<LoadingText>Fetching more posts...</LoadingText>}
```

Comment:
How about making this loading-text use `tri-dot` loading animation? I think it will look like this:
```TS
loader={<LoadingText dotsAnimated>Fetching more posts</LoadingText>}  // Note, that the text shouldn't have dots, this comment is just for you
```
In `LoadingText` simply do:
```TS
export function LoadingText({ children, dotsAnimated }: ChildrenProp & {dotsAnimated?: boolean}) {
  return <h4 className={classnamesToString(loadingText, dotsAnimated ? 'dots-animated' : undefined)}>{children}</h4>
}
```

Reply:
No need for this implementation because users won't see this Loading message. It was made just in case